app/lib/exchange-rate/exchange-rate-service.ts

```diff
@@ -27,7 +29,19 @@ class ExchangeRateService {
 
     for (const provider of providersForTickers) {
       try {
-        return await provider.getRates({ tickers, signal });
+        const rates = await provider.getRates({ tickers, signal });
+
+        // Add inverse rates
```

Comment:
I wonder if we should move this logic to each provider? Because there could be some providers that return inverse rates as well (though none of the ones we have atm does unless we make another request for other direction) and if they do I am not sure you can just safely assume it's always 1/originalRate

Also I wonder if we should return both directions if user has asked for only one ticker? Or we should only return inverse if user has asked for both tickers?

Reply:
> if they do I am not sure you can just safely assume it's always 1/originalRate

Why would it not be 1/originalRate?

> Also I wonder if we should return both directions if user has asked for only one ticker? Or we should only return inverse if user has asked for both tickers?

Yeah I think it should only return what  was requested. Makes it more obvious what is available in the returned rates too.

Reply:
> Why would it not be 1/originalRate?
> 
> >

I think there could be different spread for different direction maybe but since we are using the price for reference only probably it's fine.

If you want to keep it here instead of moving to providers lets at least put the logic into dedicated private method and also only return tickers that were asked for